frontend/packages/kubevirt-plugin/src/components/dashboards-page/overview-dashboard/activity.tsx

```diff
@@ -1,23 +1,35 @@
 import * as React from 'react';
 import * as _ from 'lodash';
-import { ActivityProgress } from '@console/shared/src/components/dashboard/activity-card/ActivityItem';
+import ActivityItem, {
+  ActivityProgress,
+} from '@console/shared/src/components/dashboard/activity-card/ActivityItem';
 import { ResourceLink } from '@console/internal/components/utils';
 import { referenceForModel } from '@console/internal/module/k8s';
 import { K8sActivityProps } from '@console/plugin-sdk';
 import { VirtualMachineModel } from '../../../models';
 
-export const DiskImportActivity: React.FC<K8sActivityProps> = ({ resource }) => (
-  <ActivityProgress
-    title="Importing VM disk"
-    progress={parseInt(_.get(resource, 'status.progress', 0), 10)}
-  >
+const VM_IMPORT_TITLE = 'Importing VM disk';
+
+export const DiskImportActivity: React.FC<K8sActivityProps> = ({ resource }) => {
+  const progress = parseInt(resource?.status?.progress, 10);
+  const vmLink = (
     <ResourceLink
       kind={referenceForModel(VirtualMachineModel)}
       name={resource.metadata.ownerReferences[0].name}
       namespace={resource.metadata.namespace}
     />
-  </ActivityProgress>
-);
+  );
+  return Number.isNaN(progress) ? (
+    <>
+      <ActivityItem>{VM_IMPORT_TITLE}</ActivityItem>
+      {vmLink}
+    </>
+  ) : (
+    <ActivityProgress title={VM_IMPORT_TITLE} progress={progress}>
```

Comment:
can we include a key?

Reply:
why ? Adding key is handled by ActivityBody https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/dashboard/activity-card/ActivityBody.tsx#L160

Reply:
ah, sorry - no need to have it here